demo/alloy-showcase-webapp/src/test/java/com/liferay/faces/test/alloy/showcase/datatable/DataTableTester.java

```diff
@@ -0,0 +1,694 @@
+			"/../../../tr[contains(@class,'info')]");
+
+		if (!browser.findElements(By.ByXPath.xpath("//input[contains(@type,'radio')]")).isEmpty()) {
+
+			controlOrCommandClick(browser, getTableRowCustomersXpath("John", "Adams"));
+		}
+		else {
+			browser.click(getTableRowCustomersXpath("John", "Adams"));
+		}
+
+		browser.waitForElementNotPresent(getTableRowCustomersXpath("John", "Adams") +
+			"/../../../tr[contains(@class,'info')]");
+		browser.click(getTableRowCustomersXpath("John", "Adams"));
+
+		browser.waitForElementVisible(getTableRowCustomersXpath("John", "Adams") +
+			"/../../../tr[contains(@class,'info')]");
+
+		// Assert UI state
+		SeleniumAssert.assertElementValue(browser, "(//table/input)", "0", false);
+
+		// Test that selected customers appear in the feedback.
+		browser.click("//button[contains(.,'Feedback')]");
+		browser.waitForElementVisible(alloyMessageXpath("John Adams"));
+
+		// Test that selected customers appear in the modal.
+		browser.click("//button[contains(.,'Modal Dialog')]");
+		browser.waitForElementTextVisible(modalDialogXpath, "John Adams");
+		SeleniumAssert.assertElementVisible(browser, modalDialogXpath);
+
+		// Assert proper feedback
+		SeleniumAssert.assertElementTextVisible(browser, modalDialogXpath + alloyOutputTextModalXpath, "John Adams");
+
+		browser.centerElementInView(modalDialogXpath + "//button[contains(.,'Cancel')]");
+		browser.click(modalDialogXpath + "//button[contains(.,'Cancel')]");
+		browser.waitUntil(ExpectedConditions.invisibilityOfElementLocated(By.ByXPath.xpath(modalDialogXpath)));
+
+		// Reset UI state
+		navigateToUseCase(browser, "dataTable", componentUseCase);
+	}
+
+	/**
+	 * Tests the Pagination feature of dataTable.
+	 */
+	protected void testPagesPaginationDefault(Browser browser) {
+
+		// test first page
+		SeleniumAssert.assertElementVisible(browser, dataTablePageAndResultsNumberXpath(1, "1-10"));
+		testDataTableDefaultPage(browser, "Elbridge", "Braxton", "Apr 21, 1713");
+
+		// test next (second) page
+		browser.clickAndWaitForAjaxRerender(dataTablePaginationLinkXpath("nextPage"));
+		SeleniumAssert.assertElementVisible(browser, dataTablePageAndResultsNumberXpath(2, "11-20"));
+		testDataTableDefaultPage(browser, "Philip", "Wolcott", "Jul 26, 1742");
+
+		// test fifth page
+		browser.clickAndWaitForAjaxRerender(dataTablePaginationLinkXpath("5"));
+		SeleniumAssert.assertElementVisible(browser, dataTablePageAndResultsNumberXpath(5, "41-50"));
+		testDataTableDefaultPage(browser, "Bernard", "Brayne", "Dec 06, 1596");
+
+		// test previous (fourth) page
+		browser.clickAndWaitForAjaxRerender(dataTablePaginationLinkXpath("previousPage"));
+		SeleniumAssert.assertElementVisible(browser, dataTablePageAndResultsNumberXpath(4, "31-40"));
+		testDataTableDefaultPage(browser, "Adoniram", "Lever", "Oct 09, 1577");
+
+		// test last (seventeenth) page
+		browser.performAndWaitForAjaxRerender(browser.createClickAction(dataTablePaginationLinkXpath("lastPage")),
```

Comment:
I know this is inconsistent with lines 665, 675, etc. because of [FACES-3006](https://issues.liferay.com/browse/FACES-3006). Let's fix that as part of this PR. Once you are done with other tasks, please let me know, and we can work on it.